### DISCUSSION TITLE
Dialog Prompt Crashes If Requesting a Second One

### DISCUSSION THREAD

**Discussion by cryptoKevinL** [MetaMask/snaps#1490] [Q&A]
When a Dialog Prompt is already open, if my snap requests to do another Dialog Prompt, the current one crashes with an error:
<img width="600" alt="Screenshot 2023-06-09 at 11 51 54 AM" src="https://github.com/MetaMask/snaps/assets/19207330/15e3d357-ef38-4c7e-ae4c-b94afc94aa7f">

Is there a way we should be requesting a pending dialog state from some official Snaps function, or for now should Snaps owners just keep a global state/mutex around this?

**Comment by cryptoKevinL** [comment]
@ziad-saab - any clarification you can provide here?

**Comment by cryptoKevinL** [comment]
I see this in the example snaps for protection, but I can't see to find where the text actually comes from:
<img width="433" alt="Screenshot 2023-06-19 at 9 24 15 PM" src="https://github.com/MetaMask/snaps/assets/19207330/462bd093-7332-469a-a172-a7db56fd954f">

**Comment by cryptoKevinL** [comment]
I also get this consistently when running the example app, not sure if this is important or not
<img width="1302" alt="Screenshot 2023-06-19 at 9 43 28 PM" src="https://github.com/MetaMask/snaps/assets/19207330/ee2203fc-c287-4731-b3f3-18bcfa95fb81">

**Comment by Montoya** [comment]
I will look into this issue but you should be `await`ing the result of the dialog request. Every request is a promise that will resolve.

**Reply by cryptoKevinL** [reply]
thanks @Montoya - we do this: 
<img width="449" alt="Screenshot 2023-06-22 at 10 34 05 AM" src="https://github.com/MetaMask/snaps/assets/19207330/24d926c8-0124-4c3f-aa46-f20a311ffc7c">

But, this is fired from within a cron job, so I'm guessing the cron job takes priority.  I've added a simple mutex and seems to get us around this issue for now.

**Reply by Montoya** [reply]
Ah I see, yes in the cronjob it's not so trivial
